Test/VimCoreTest/NormalModeIntegrationTest.cs

```diff
@@ -2734,6 +2734,91 @@ public void Issue1266()
+            }
+
+            /// <summary>
+            /// Undo of insert with backspaces
+            /// </summary>
+            [WpfFact]
+            public void Undo_InsertWithBackspaces()
+            {
+                Create("aaa bbb ccc");
+                _vimBuffer.ProcessNotation("1Gwiddd<BS><BS>ef <Esc>");
+                Assert.Equal("aaa def bbb ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa bbb ccc", _textView.GetLine(0).GetText());
+            }
+
+            /// <summary>
+            /// Undo of insert with arrow keys
+            /// </summary>
+            [WpfFact]
+            public void Undo_InsertWithArrowKeys()
+            {
+                Create("aaa bbb ccc");
+                _vimBuffer.ProcessNotation("1Gwieee <Left><Left><Left><Left>ddd <Esc>");
+                Assert.Equal("aaa ddd eee bbb ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa eee bbb ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa bbb ccc", _textView.GetLine(0).GetText());
+            }
+
+            /// <summary>
+            /// Undo of insert with a break in the undo sequence
+            /// </summary>
+            [WpfFact]
+            public void Undo_InsertWithBreakUndoSequence()
+            {
+                Create("aaa bbb ccc");
+                _vimBuffer.ProcessNotation("1Gwiddd <C-g>ueee <Esc>");
+                Assert.Equal("aaa ddd eee bbb ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa ddd bbb ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa bbb ccc", _textView.GetLine(0).GetText());
+            }
+
+            /// <summary>
+            /// Undo of change word
+            /// </summary>
+            [WpfFact]
+            public void Undo_ChangeWord()
+            {
+                Create("aaa bbb ccc");
+                _vimBuffer.ProcessNotation("1Gwcwddd<Esc>");
+                Assert.Equal("aaa ddd ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa bbb ccc", _textView.GetLine(0).GetText());
+            }
+
+            /// <summary>
+            /// Undo of change word with backspaces
+            /// </summary>
+            [WpfFact]
+            public void Undo_ChangeWordWithBackspaces()
+            {
+                Create("aaa bbb ccc");
+                _vimBuffer.ProcessNotation("1Gwcwddd<BS><BS>ef<Esc>");
+                Assert.Equal("aaa def ccc", _textView.GetLine(0).GetText());
+                _vimBuffer.ProcessNotation("u");
+                Assert.Equal("aaa bbb ccc", _textView.GetLine(0).GetText());

```

Comment:
Can we add some tests for the up and down arrow keys as well? Those are the other case where I know of that the undo sequence can get broken up.

Reply:
Sure, I added one test that uses `<C-Left>` and another that uses `<Down>`. The implementation detects edit vs. non-edit using the flag specified in the key binding, i.e. whether `CommandFlags.InsertEdit` is present. None of the arrow keys or arrow keys with modifiers have that flag set.